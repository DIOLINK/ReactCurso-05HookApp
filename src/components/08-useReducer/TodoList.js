import React from 'react';
import PropTypes from 'prop-types';
import TodoListItem from './TodoListItem';

const TodoList = ({ todos, handleToggle, handleDelete }) => {
  return (
    <ul className="list-group list-group-flush">
      {todos.map((todo, i) => (
        <TodoListItem
          todo={todo}
          handleToggle={handleToggle}
          handleDelete={handleDelete}
          index={i}
          key={todo.id}
        />
      ))}
    </ul>
  );
};
TodoList.propTypes = {
  todos: PropTypes.array.isRequired,
  handleToggle: PropTypes.func.isRequired,
  handleDelete: PropTypes.func.isRequired,
};
export default TodoList;
