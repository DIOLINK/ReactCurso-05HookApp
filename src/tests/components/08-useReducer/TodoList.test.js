import { shallow } from 'enzyme';
import TodoList from '../../../components/08-useReducer/TodoList';
import { dummyTodo } from '../../fixtures/dummyTodo';

describe('Prueba en <TodoList />', () => {
  const handleToggle = jest.fn();
  const handleDelete = jest.fn();
  const wrapper = shallow(
    <TodoList
      todos={dummyTodo}
      handleToggle={handleToggle}
      handleDelete={handleDelete}
    />
  );
  it('Debe de mostrar el componente', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('Debbe de tener dos <TodoListItem />', () => {
    expect(wrapper.find('TodoListItem').length).toBe(dummyTodo.length);
  });
});
