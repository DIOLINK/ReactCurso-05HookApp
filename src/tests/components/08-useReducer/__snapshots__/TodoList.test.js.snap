// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <TodoList /> Debe de mostrar el componente 1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={0}
    key="1"
    todo={
      Object {
        "description": "Hacer la compra",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={1}
    key="2"
    todo={
      Object {
        "description": "Pasear al perro",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;
